File: Zamba/ZambaWeb2013/zamba2010/Zamba.Filters/IFiltersComponent.cs
using System;
using System.Collections.Generic;
using Zamba.Core;

namespace Zamba.Filters
{
    public interface IFiltersComponent
    {
        List<FilterElem> GetLastUsedFilters(Int64 docTypeId, Int64 currentUserId, bool IsTask);
        Int32 GetDocumentFiltersCount(Int64 docTypeId, bool IsTask);
        FilterElem SetNewFilter(Int64 indexId, String filterName, IndexDataType dataType, Int64 currentUserId,
                                                String compareOperator, String valueString, Int64 docTypeId, Boolean save, String description, IndexAdditionalType additionalType, String FilterType, Boolean IsTask);
        void SaveFilterInDatabase(IFilterElem fe);
        void SetEnabledFilter(IFilterElem fe, Boolean IsTask);
        void RemoveFilter(IFilterElem fe, Boolean IsTask);
        void ClearFilters(Int64 docTypeId, Int64 userId, Boolean IsTask, System.Data.DataTable dataTable, Boolean RemoveFilterRightz);
    }
}

File: Zamba/zambawindows2013/Zamba2010/Zamba.ICore/IFiltersComponent.vb
﻿Public Interface IFiltersComponent
    Function GetLastUsedFilters(ByVal docTypeId As Int64, ByVal currentUserId As Int64, ByVal filterType As FilterTypes) As List(Of IFilterElem)
    Function GetDocumentFiltersCount(ByVal docTypeId As Int64, ByVal filterType As FilterTypes) As Int32
    Function SetNewFilter(ByVal indexId As Int64, ByVal filterName As String, ByVal dataType As IndexDataType, ByVal currentUserId As Int64, ByVal compareOperator As String, ByVal valueString As String, ByVal docTypeId As Int64, ByVal save As Boolean, ByVal description As String, ByVal additionalType As IndexAdditionalType, ByVal FilterType As String, ByVal FType As FilterTypes) As IFilterElem
    Sub SaveFilterInDatabase(ByVal fe As IFilterElem)
    Sub SetEnabledFilter(ByVal fe As IFilterElem, ByVal filterType As FilterTypes)
    Sub RemoveFilter(ByVal fe As IFilterElem, ByVal filterType As FilterTypes)
    Sub ClearFilters(ByVal docTypeId As Int64, ByVal userId As Int64, ByVal filterType As FilterTypes, ByVal dataTable As DataTable, ByVal RemoveFilterRightz As Boolean)
End Interface